src/main/java/brokeculator/parser/SummariseParser.java

```diff
@@ -4,61 +4,97 @@
 import brokeculator.command.InvalidCommand;
 import brokeculator.command.SummariseCommand;
 
+import java.time.LocalDateTime;
+
 public class SummariseParser {
-    private static final String INVALID_FORMAT_MESSAGE = "Invalid input format for summarise command. "
-            + "Please use the format: summarise /from <start> /to <end>";
+    private static final String[] SUMMARISE_COMMAND_OPTIONS = {" /n ", " /d ", " /c ", " /from ", " /to "};
+    private static final int NAME_INDEX = 0;
+    private static final int DATE_INDEX = 1;
+    private static final int CATEGORY_INDEX = 2;
+    private static final int FROM_INDEX = 3;
+    private static final int TO_INDEX = 4;
 
     public static Command parseInput(String userInput) {
         String[] userInputAsArray = userInput.trim().split("\\s+");
+        String nameToSummariseBy = null;
+        LocalDateTime dateToSummariseBy = null;
+        String categoryToSummariseBy = null;
         int beginIndex = 0;
         int endIndex = -1;
-        if (userInputAsArray.length == 1) {
-            return new SummariseCommand(beginIndex, endIndex);
+
+        if (userInput.contains(SUMMARISE_COMMAND_OPTIONS[NAME_INDEX])) {
+            nameToSummariseBy = getOptionField(userInputAsArray, SUMMARISE_COMMAND_OPTIONS[NAME_INDEX]);
+            nameToSummariseBy = nameToSummariseBy.isBlank() ? null : nameToSummariseBy;
         }
-        if (userInputAsArray.length == 3) {
-            if (userInputAsArray[1].equals("/from")) {
-                try {
-                    beginIndex = Integer.parseInt(userInputAsArray[2]) - 1;
-                } catch (NumberFormatException e) {
-                    return new InvalidCommand("Start index cannot be non-integer");
-                }
-                if (beginIndex < 0) {
-                    return new InvalidCommand("Start index must be one or greater");
-                }
-                // endIndex == -1;
-                return new SummariseCommand(beginIndex, endIndex);
-            } else if (userInputAsArray[1].equals("/to")) {
-                // beginIndex = 0;
-                try {
-                    endIndex = Integer.parseInt(userInputAsArray[2]) - 1;
-                } catch (NumberFormatException e) {
-                    return new InvalidCommand("End index cannot be non-integer");
-                }
-                if (endIndex < 0) {
-                    return new InvalidCommand("End index must be one or greater");
-                }
-                return new SummariseCommand(beginIndex, endIndex);
+
+        // TODO implement date processing
+
```

Comment:
we might need to determine whether the summarize command conditions are && or ||